### ISSUE TITLE
$RESTIC_PASSWORD_COMMAND has no effect

### ISSUE THREAD

**Issue by Myridium** [restic/restic#3585]
Output of `restic version`
--------------------------
`restic 0.12.1 compiled with go1.17.3 on linux/amd64`

How did you run restic exactly?
-------------------------------
```
RESTIC_PASSWORD_COMMAND="pass restic"
restic check --read-data -r .
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local filesystem.

Expected behavior
-----------------
Something like

```
using temporary cache in /tmp/restic-check-cache-2302040508
enter password for repository: 
repository ef829401 opened successfully, password is correct
created new cache in /tmp/restic-check-cache-2302040508
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
[0:00] 100.00%  23 / 23 snapshots
read all data
[0:00] 100.00%  51 / 51 packs
no errors were found
```

Actual behavior
---------------
```
using temporary cache in /tmp/restic-check-cache-2348329897
enter password for repository: 
```

Steps to reproduce the behavior
-------------------------------
```
mkdir -p ~/tmp/restic
cd ~/tmp/restic
restic init -r . <<< "password1" 
RESTIC_PASSWORD_COMMAND="echo password1"
restic check -r .
```

Output is:
```
using temporary cache in /tmp/restic-check-cache-3039215301
enter password for repository:
```

Do you have any idea what may have caused this?
-----------------------------------------------
No.

Do you have an idea how to solve the issue?
-------------------------------------------
Review `restic` code for where `$RESTIC_PASSWORD_COMMAND` is supposed to be used. 

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
No, it has been a pain in the backside. Blank password support has been refused for apparently ideological reasons. And  the `$RESTIC_PASSWORD_COMMAND` variable does not work. I do not want to use `restic check -r . <<< "$(password_command)"` every time. There is no good reason that backup software should enforce a password.

**Comment by ibib** [comment]
If I get your problem right you should be using `RESTIC_PASSWORD` instead of `RESTIC_PASSWORD_COMMAND`.

**Comment by Myridium** [comment]
@ibib from the `restic --help`:

```
      --password-command command   shell command to obtain the repository password from (default: $RESTIC_PASSWORD_COMMAND)
```

**Comment by MichaelEischer** [comment]
@Myridium Did you `export` the environment variable such that you shell actually passes it on to restic?

**Comment by Myridium** [comment]
> @Myridium Did you `export` the environment variable such that your shell actually passes it on to restic?

No, I did not. Thank you for pointing that out. 

This issue is user error; closing.

Correct usage:
```
mkdir -p ~/tmp/restic
cd ~/tmp/restic
restic init -r . <<< "password1" 
export RESTIC_PASSWORD_COMMAND="echo password1"
restic check -r .
```

However, I do have another suggestion. **The user can set the wrong password without realising** if they pipe `stdin` into `restic` while having `RESTIC_PASSWORD_COMMAND` set. I just did this myself while testing this updated usage example. Since I don't think `restic init` uses `stdin` for anything except accepting a password, I think it would be safest to **abort if anything is received on `stdin` while `RESTIC_PASSWORD_COMMAND` or `RESTIC_PASSWORD_FILE` are set, or if both `RESTIC_PASSWORD_COMMAND` and `RESTIC_PASSWORD_FILE` are set, regardless of `stdin`.**
